app/models/services.py

```diff
@@ -12,67 +15,123 @@ class ErrInvalidParameters(Exception):
-    query = tinydb.Query()
-    services.remove(query.service_id == service_id)
-
-
-def all():
-    results = services.all()
-
-    if not results:
-        raise ErrNotFound("oops.. no service found in the db")
-
-    return results
-
-
-def get(service_id):
-    service = tinydb.Query()
-    result = services.search(service.service_id == service_id)
-
-    if not result:
-        raise ErrNotFound(f"no service found for the provided id: {service_id}")
-
-    return result[0]
-
-
-def get_by_name(name=""):
-    if not name:
-        raise ErrInvalidParameters("service name must be provided")
-
-    service = tinydb.Query()
-    result = services.search(service.name == name)
-
-    if not result:
-        raise ErrNotFound(f"no service found for the provided name: {name}")
-
-    return result[0]
+class ServiceRepo:
+
+    def __init__(self, db: tinydb.TinyDB):
+        self.db = db
+        self.services = db.table('services')
+
+    def insert(self, id="",
+               shop_id="",
+               name="",
+               slug="",
+               description="",
+               price="",
+               images=None):
+
+        doc = {"shop_id": shop_id,
+               "id": id,
+               "name": name,
+               "slug": slug,
+               "description": description,
+               "price": price,
+               "images": images if images is not None else []}
+        self.services.insert(doc)
+        return id
+
+    def update(self, id="",
+               name="",
+               slug="",
+               description="",
+               price="",
+               images=None):
+        query = tinydb.Query()
+        doc = {"name": name,
+               "slug": slug,
+               "description": description,
+               "price": price,
+               "images": images if images is not None else []}
+        self.services.update(doc, query.id == id)
+        return id
+
+    def delete(self, id):
+        query = tinydb.Query()
+        self.db.table('services').remove(query.id == id)
+
+    def all(self):
+        results = self.services.all()
+
+        if not results:
+            raise ErrNotFound("oops.. no service found in the db")
+
+        return [Service(**doc) for doc in results]
+
+    def get(self, id: str):
+        service = tinydb.Query()
+        result = self.services.search(service.id == id)
+
+        if not result:
+            raise ErrNotFound(f"no service found for the provided id: {id}")
+
+        return Service(**result.pop())
+
+    def get_by_name(self, name=""):
+        if not name:
+            raise ErrInvalidParameters("service name must be provided")
+
+        service = tinydb.Query()
+        result = self.services.search(service.name == name)
+
+        if not result:
+            raise ErrNotFound(f"no service found for the provided name: {name}")
+
+        return result[0]
+
+
+class Service:
+    UPLOADED_DIR_PATH: str = str(Path.cwd() / Path("app/static/uploaded"))
```

Comment:
For the future, this should be placed in the config.py file (but don't worry about it from now)